controller/engine_image_controller.go

```diff
@@ -283,7 +284,19 @@ func (ic *EngineImageController) syncEngineImage(key string) (err error) {
 	if err != nil {
 		return errors.Wrapf(err, "cannot get daemonset for engine image %v", engineImage.Name)
 	}
+
+	isReady := true
```

Comment:
For simplicity, can we just clean up the corresponding engine manager and replica manager if the daemonset pod is being deleted/not running?

Setting the engine image to state `deploying` is fine. It just blocks creating new volumes with this engine image as long as we don't clean up all instance managers.

Reply:
Can you clarify the first part? Would it affect the toleration case?

Reply:
When users modify the toleration setting, the old daemonset pod will be deleted. But it is update rather than deletion, all instance manager won't be deleted by the delete propagation. Then the old instance manager won't be deleted and the new tolerations won't be applied to the instance manager pods. 

That's one main motivation for this instance manager cleanup.

Reply:
As I said, we can take `torleration` in the reconciliation consideration to fix this issue.